grad_login/lib/infrastructure/medicine/interactions_service.dart

```diff
@@ -23,7 +24,16 @@ class InteractionsService {
         "Authorization": "JWT $token",
       },
     );
-    final responseData = json.decode(response.body);
+    if (response.statusCode == 500) {
```

Comment:
you can simplify this into breaking early.
```Dart
if(response.statusCode == 500){
   // your logic
   return responseData;
}
responseData = json.decode(response.body);
if (responseData.isEmpty) {
   // your logic
}

return responseData
```
in this way, you're avoiding nested if chain.
    


```